data_warehouse/decorators/context_params_check_decorators.py

```diff
@@ -0,0 +1,42 @@
+'''context参数检查装饰器模块'''
+from infra.exceptions import BadRequest
+import functools
+
+
+def __check_request(args):
+    if args[-1] is None:
+                raise BadRequest('请给定context参数。')
+    context = args[-1]
+    if context.get('request', None) is None:
+        raise BadRequest('未在context参数中指定request。')
+
+
+def request_required(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        __check_request(args)
+        return func(*args, **kwargs)
+
+
+def admin_required(mode='department_admin'):
+    def actual_decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            __check_request(args)
```

Comment:
you might consider returning the request if check succeed, then you don’t have to extract it again